### PR TITLE
E2SM-RC adding pdubuilder for some of the PDUs

### PR DESCRIPTION
**Pull request by eroshiva**

### CODE REVIEW

File: servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			RanFunctionE2SmOid:     oID,
+			RanFunctionDescription: dscr,
+		},
+		RanFunctionDefinitionControl: control,
+	}
+
+	return msg, nil
+}
+
+func CreateE2SmRcRanfunctionDefinitionPolicy(rfsn string, oID string, dscr string, policy *e2smrcv1.RanfunctionDefinitionPolicy) (*e2smrcv1.E2SmRcRanfunctionDefinition, error) {
```
Comment (adibrastegarnia) [inline servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go:81]: We should be able to set all Policy, Control, Insert definitions part of one ran function description. Why they are separate? Are you using builder patterns?

File: servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			RanFunctionShortName:   rfsn,
+			RanFunctionE2SmOid:     oID,
+			RanFunctionDescription: dscr,
+		},
+	}
+
+	return msg, nil
+}
+
+func CreateE2SmRcRanfunctionDefinitionEventTrigger(rfsn string, oID string, dscr string, et *e2smrcv1.RanfunctionDefinitionEventTrigger) (*e2smrcv1.E2SmRcRanfunctionDefinition, error) {
```
Comment (adibrastegarnia) [inline servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go:25]: I would suggest to use complete variable name. oID should OID but rfsn should ranFunctionShortName. It is not clear what this abbreviation stands for.

File: servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go (hunk 1 above)
Comment (adibrastegarnia) [reply servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go:81]: Would be nice to have a unit test which shows how to build a complete ran function description which can be included in E2 setup and it is consistent with encoding of optional/non optional fields (i.e. the minimum that we should set that we can include it in E2 setup. )

Comment (adibrastegarnia) [reply servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go:81]: Look at this example in ran-simulator for kpmv2 https://github.com/onosproject/ran-simulator/blob/master/pkg/utils/e2sm/kpm2/ranfuncdescription/ranfuncdescription.go#L82

Comment (eroshiva) [reply servicemodels/e2sm_rc/pdubuilder/E2SM-RcRanfunctionDefinition.go:81]: Agree. All lists included in the RanfunctionDescription are optional, so I assume that their presence may vary depending on the use-case (i.e., event trigger, report, policy, etc.). 

Builder pattern for RFD to come in next push. For all other SMs to come in the next PR to unblock you.
Unit tests are also to come in the next PR.

**Review by adibrastegarnia — APPROVED** [review]

**Pull request merged**